### PR TITLE
[Debt] Implements `useBreadcrumbs` hook broadly

### LINKED ISSUE #9764: ♻️ Implement `useBreadcrumbs` hook
## ♻️ Refactor

Change all breadcrumbs to use `useBreadcrumbs` hook.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/3d2d51b3cd328564e7707e0e4879939b85341331/apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx#L155-L160

## 🕵️ Details

This would eliminate the need to include the home crumb that added automatically in `useBreadcrumbs` hook.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/3d2d51b3cd328564e7707e0e4879939b85341331/apps/web/src/pages/Skills/TopBehaviouralSkillsPage.tsx#L51-L59

## 🙋‍♀️ Proposed Solution

- Change all breadcrumbs to use `useBreadcrumbs` hook
- add `isAdmin` prop to `useBreadcrumbs` hook for verifying admin to link to `adminDashboard` route

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt was addressed and expected functionality has not been affected.

- [ ] all breadcrumbs use `useBreadcrumbs` hook
- [ ] `isAdmin` prop exists on `useBreadcrumbs` hook for verifying admin to link to `adminDashboard` route

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #9764.

## 👋 Introduction

This PR implements the `useBreadcrumbs` hook on all breadcrumbs and adds the `isAdmin` prop so it can be used on admin pages, where possible. It also consolidates the **Home** string to avoid potential diverging translations.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Verify breadcrumbs do not include **Home** within crumbs array in `useBreadcrumbs` object
2. Verify breadcrumbs appear as they did before and link to the proper page

### CODE REVIEW

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx [hunk 1]
```diff
@@ -231,28 +232,23 @@ export const AssessmentPlanBuilderPage = () => {
 
   // Note: Should technically be in subNav of layout?
```
Comment (mnigh) [inline apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx:233]: @esizer I updated this but `navigationCrumbs` is an unused variable. does it need to remain in the code or can it be removed entirely?

Comment (esizer) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx:233]: Yeah, you can :axe: this. Those crumbs are defined in the layout component.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/aed46eb6ef5113587503145445582f08de28b495/apps/web/src/utils/poolUtils.tsx#L317-L347

Comment (mnigh) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx]: [b463e04](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9796/commits/b463e0473b4a2c21f01af4b6ece2af259d4ab6c0)

**Review by esizer — APPROVED** [review]
LGTM! :+1: 

Even if chromatic was being silly :face_with_spiral_eyes:

**Pull request merged**